sdk/storage/azure-storage-file/src/main/java/com/azure/storage/file/FileServiceSasSignatureValues.java

```diff
@@ -394,53 +389,85 @@ public FileServiceSasSignatureValues setContentType(String contentType) {
      * Uses an account's shared key credential to sign these signature values to produce the proper SAS query
      * parameters.
      *
+     * <p>
+     * If {@link #getVersion()} is not set, the {@link FileServiceVersion#getLatest() latest service version} is used.
+     *
+     * </p>
+     * <p>
+     * The type of SAS query parameters returned depends on the following:
+     * <ol>
+     *     <li>If {@link #getFilePath()} is not set, query parameters for a <b>share SAS</b> are returned.</li>
+     *     <li>Otherwise, {@link #getShareName()} and {@link #getFilePath()} are used to create query parameters for a
+     *     <b>file SAS</b>.</li>
+     * </ol>
+     *
+     * See class level JavaDocs for code snippets.
+     *
      * @param storageSharedKeyCredentials A {@link StorageSharedKeyCredential} object used to sign the SAS values.
      * @return {@link FileServiceSasQueryParameters}
      * @throws IllegalStateException If the HMAC-SHA256 algorithm isn't supported, if the key isn't a valid Base64
      * encoded string, or the UTF-8 charset isn't supported.
-     * @throws NullPointerException If {@code storageSharedKeyCredentials} is null. Or when any of {@code version},
-     * {@code canonicalName} or {@code resource} is null. Or if {@code identifier} is not set and any of
-     * {@code expiryTime} or {@code permissions} is null. Or if {@code expiryTime} and {@code permissions} are not set
-     * and {@code identifier} is null
+     * @throws IllegalArgumentException if {@link #getPermissions()} contains an invalid character for the SAS resource.
+     * @throws NullPointerException If {@code storageSharedKeyCredentials} is null.
      */
-    public FileServiceSasQueryParameters generateSASQueryParameters(
+    public FileServiceSasQueryParameters generateSasQueryParameters(
         StorageSharedKeyCredential storageSharedKeyCredentials) {
         Utility.assertNotNull("storageSharedKeyCredentials", storageSharedKeyCredentials);
-        assertGenerateOK();
+
+        final String resource;
+        if (ImplUtils.isNullOrEmpty(filePath)) {
+            resource = Constants.UrlConstants.SAS_SHARE_CONSTANT;
+
+            // Make sure the permission characters are in the correct order
+            if (permissions != null) {
+                permissions = ShareSasPermission.parse(permissions).toString();
+            }
+        } else {
+            resource = Constants.UrlConstants.SAS_FILE_CONSTANT;
+
+            // Make sure the permission characters are in the correct order
+            if (permissions != null) {
+                permissions = FileSasPermission.parse(permissions).toString();
```

Comment:
If someone passed a `ShareSasPermission` with list and attempts to generated a File SAS this will throw, and I know this is called out in the docs, but is it the most user friendly way to go about this?

Reply:
I can call it out in docs, but its the same behaviour as the other languages.